### PR TITLE
SW-2427 Intermediate effort of reassignment flow

### PR DESCRIPTION
**Pull request by karthikbtf**
- api wrappers
- table with partial info
- basic styling etc.

Terraware App 2022-12-06 15-25-33

Terraware App 2022-12-06 15-25-49

### CODE REVIEW

File: src/components/common/TitleDescription.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { Box, Typography, useTheme } from '@mui/material';
```
Comment (karthikbtf) [inline src/components/common/TitleDescription.tsx:1]: I created this so we can reuse as needed, avoid a lot of copy pasted code.

File: src/components/common/Card.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { ReactNode } from 'react';
```
Comment (karthikbtf) [inline src/components/common/Card.tsx:1]: created this so we can reuse this card as needed, avoid some boilerplate code

File: src/components/NurseryWithdrawals/NurseryReassignment.tsx [hunk 3]
```diff
@@ -1,13 +1,193 @@
...
+  }, [deliveryId, snackbar]);
+
+  // populate lookup maps of zones and plots
+  useEffect(() => {
+    if (!delivery) {
+      return;
+    }
+
+    const populatePlots = async () => {
+      const plotIds = delivery.plantings.map((planting) => planting.plotId).filter((id) => id);
```
Comment (constanzauanini) [inline src/components/NurseryWithdrawals/NurseryReassignment.tsx:92]: What is filter id=> id doing ?

Comment (karthikbtf) [reply src/components/NurseryWithdrawals/NurseryReassignment.tsx:92]: plantings can have undefined plotid, we filter only those ids that are valid

File: src/components/NurseryWithdrawals/NurseryReassignment.tsx [hunk 4]
```diff
@@ -1,13 +1,193 @@
...
+          originalPlot: zones[planting.plotId]?.plotName,
+        };
+      });
+  }, [delivery, siteName, speciesMap, zones]);
+
+  const reassignmentRenderer = useMemo(
+    () =>
+      ReassignmentRenderer({
+        plots: plots || {},
+        zones: zones || {},
```
Comment (constanzauanini) [inline src/components/NurseryWithdrawals/NurseryReassignment.tsx:162]: Why do you need the zones in the table?

File: src/components/NurseryWithdrawals/NurseryReassignment.tsx (hunk 3 above)
Comment (constanzauanini) [reply src/components/NurseryWithdrawals/NurseryReassignment.tsx:92]: Thanks!

File: src/components/NurseryWithdrawals/NurseryReassignment.tsx (hunk 4 above)
Comment (karthikbtf) [reply src/components/NurseryWithdrawals/NurseryReassignment.tsx:162]: I don't know yet. This is intermediate, I will remove it later if completely not needed.

File: src/components/NurseryWithdrawals/NurseryReassignment.tsx [hunk 5]
```diff
@@ -1,13 +1,193 @@
...
+      <TitleDescription title={strings.REASSIGN_SEEDLINGS} description={strings.REASSIGN_SEEDLINGS_DESCRIPTION} />
+      {zones ? (
+        <Box
+          paddingBottom={isMobile ? '185px' : '105px'}
+          display='flex'
+          flexDirection={isMobile ? 'row' : 'column'}
+          marginTop={theme.spacing(3)}
+        >
+          <Card>
+            <Table
```
Comment (constanzauanini) [inline src/components/NurseryWithdrawals/NurseryReassignment.tsx:178]: We can add showPagination={false}

Comment (karthikbtf) [reply src/components/NurseryWithdrawals/NurseryReassignment.tsx:178]: sounds good - will add it, thanks.

**Review by constanzauanini — APPROVED** [review]
Looks good! One suggestion on the table!

**Comment by karthikbtf** [comment]
> Looks good! One suggestion on the table!

Thanks @constanzauanini - I will make the changes.

**Pull request merged**